x/evm/keeper/state_transition.go

```diff
@@ -392,7 +414,7 @@ func (k *Keeper) ApplyMessageWithConfig(ctx sdk.Context,
 	// access list preparation is moved from ante handler to here, because it's needed when `ApplyMessage` is called
 	// under contexts where ante handlers are not run, for example `eth_call` and `eth_estimateGas`.
 	if rules := cfg.ChainConfig.Rules(big.NewInt(ctx.BlockHeight()), cfg.ChainConfig.MergeNetsplitBlock != nil); rules.IsBerlin {
-		stateDB.PrepareAccessList(msg.From(), msg.To(), vm.ActivePrecompiles(rules), msg.AccessList())
+		stateDB.PrepareAccessList(msg.From(), msg.To(), vm.DefaultActivePrecompiles(rules), msg.AccessList())
```

Comment:
By declaring specific storage slots in advance, access list of evm reduces the gas cost of accessing those slots. Similar to other precompiled contracts, app-level precompile contracts will almost always benefit from gas cost savings because they will be called as cross-contract calls most of the time. This is why i changed to `evm.ActivePrecompiles` instead of `vm.DefaultActivePrecompiles`.